### DISCUSSION TITLE
Can not render a <pre> element inside a <table>

### DISCUSSION THREAD

**Discussion by jgthms** [mdx-js/mdx#1910] [Q&A] [answered]
Hi I'm trying to render a `<pre>` block inside a `<td>` but it's not rendering (I'm using mdx within Storybook).

Example code:

```mdx
<table>
  <tbody>
    <tr>
      <td>
        <pre>.button {
  color: blue;
}</pre>
      </td>
    </tr>
  </tbody>
</table>
```

Is there a way to render a code block inside a table?

**Comment by jgthms** [comment]
This seems to work, but it strips the whitespace, which should be preserved in `<pre>` elements:

```
<table>
  <tbody>
    <tr>
      <td>
        <pre>.button \{
  color: blue;
\}</pre>
      </td>
    </tr>
  </tbody>
</table>
```

**Comment by ChristianMurphy** [comment, marked as answer]
It looks like you are trying to render a codeblock?
Another option would be try something like:

````jsx
<table>
<tbody>
<tr>
<td>

```css 
.button {
  color: blue;
}
```

</td>
</tr>
</tbody>
</table>
````

mdx 1 (currently  used by storybook) allows markdown to be embedded in JSX if there are spacing newlines between the JSX and the Markdown.

**Comment by jgthms** [comment]
Thanks that's helpful. I had to remove all indentation in the whole table like you did. Unfortunate requirement but it works.

**Reply by wooorm** [reply]
You’re trying to type as HTML.
But this project is MDX: Markdown (- HTML) + JSX. You have to escape `{` inside JSX.
More info in the docs: https://mdxjs.com/docs/what-is-mdx/.
